### PR TITLE
Add pytest and .prow.yaml

### PR DESCRIPTION
**Pull request by xtuchyna**
## Related Issues and Dependencies
Add pytest

### CODE REVIEW

File: .aicoe-ci.yaml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 check:
+  - thoth-pytest38
   - thoth-precommit
```
Comment (harshad16) [inline .aicoe-ci.yaml:3]: Remove these two

File: .prow.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      containers:
+        - image: quay.io/thoth-station/thoth-precommit-py38:v0.12.5
+          command:
+            - "pre-commit"
+            - "run"
+            - "--all-files"
+          resources:
+            requests:
+              memory: "2Gi"
+              cpu: "2"
```
Comment (harshad16) [inline .prow.yaml:17]: Insert follow snippet for pytest 
```
  - name: thoth-pytest-py38
    decorate: true
    skip_report: false
    always_run: true
    context: aicoe-ci/prow/pytest
    spec:
      containers:
        - image: quay.io/goern/thoth-pytest:v0.13.0-dev
          command:
             - "/bin/run-pytest"
          resources:
            requests:
              memory: "2Gi"
              cpu: "2"
```

**Review by harshad16 — CHANGES_REQUESTED** [review]
/lgtm

left some minor changes to be done.

File: .prow.yaml (hunk 2 above)
Comment (xtuchyna) [reply .prow.yaml:17]: :100: !

**Pull request merged**